### PR TITLE
rv_sample expect bug

### PR DESCRIPTION
**Pull request by ev-br**
#### Reference issue

Reported at Stackoverflow, https://stackoverflow.com/questions/63199792/why-do-scipy-stats-rv-discrete-mean-and-expect-show-unexpected-difference

`rv_sample.expect` gives wrong answers in some cases.

#### What does this implement/fix?

`rv_discrete.expect` iterates over the support of the distribution, assuming that it contains contiguous integers (This is baked into discrete distributions, save `rv_sample`.)  Thus chunked iteration in  `_expect` gives wrong results if the sample points  `xk` are not equidistant with `self.inc == 1`. Furthermore, the bug is masked by a shortcut in `_expect` for small lengths of `rv_sample.xk`

So this PR just reimplements `_expect` for `rv_sample`.

#### Additional information

Also remove an `_argcheck` call which is no longer needed now that `_argcheck` is not allowed to set `.a` and `.b`

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -3470,7 +3470,6 @@ def fun(x):
         # might be problems(?) with correct self.a, self.b at this stage maybe
```
Comment (sethtroisi) [inline scipy/stats/_distn_infrastructure.py:3470]: nit: does the docstring having a typo?

```
 lb, ub : int, optional
            Lower and upper bound for the summation, default is set to the
            support of the distribution, inclusive (``ul <= k <= ub``).	            
```
should that be (``**lb** <= k <= ub``).

**Review by sethtroisi — APPROVED** [review]

File: scipy/stats/_distn_infrastructure.py (hunk 1 above)
Comment (ev-br) [reply scipy/stats/_distn_infrastructure.py:3470]: Nice catch. The typo seems to have been there forever (tm). Fixed in the last commit.

**Comment by sethtroisi** [comment]
Thanks @ev-br, It's amazing that you proactively search out issues on StackOverflow.

**Comment by ev-br** [comment]
I don't proactively monitor Stackoverflow. When I'm there (usually because it popped up in a google search), I skim over the [scipy] tag before closing the tab, that's all.

**Pull request merged**